syntax = "proto2";

message ShmMsg {
	required int32 tag = 1;
	
	oneof value {
		int32 intValue = 2;
		float floatValue = 3;
		bool boolValue = 4;
	}
	// If a value is not present, the message is a variable read request
}
